modules/library/src/main/scala/zio/elasticsearch/ElasticAggregation.scala

```diff
@@ -353,4 +353,53 @@ object ElasticAggregation {
   final def valueCountAggregation(name: String, field: String): ValueCountAggregation =
     ValueCount(name = name, field = field)
 
+  /**
+   * Constructs a type-safe instance of [[zio.elasticsearch.aggregation.WeightedAvgAggregation]] using the specified
+   * parameters.
+   *
+   * @param name
+   *   aggregation name
+   * @param valueField
+   *   the type-safe field for value for which weighted avg aggregation will be executed
+   * @param weightField
+   *   the type-safe field for weight for which weighted avg aggregation will be executed
```

Comment:
Maybe something like this:
```suggestion
   *   the type-safe field that represents weight for which weighted avg aggregation will be executed
```

Reply:
Something like this for `valueField` also.